### ISSUE TITLE
Hosts addon does not work on Docker image

### ISSUE THREAD

**Issue by jrjohnson** [travis-ci/travis-ci#2969]
I'm suddenly unable to bind to port 4200.  I get:

*Could not serve on http://localtest:4200. It is either in use or you do not have permission.*

I need to be able to start a node express server (via the command ember serve) so my tests have access to the API.  I attempted to use sudo, but then I have path issues with node because I gather using nvm it is installed for the travis user and not root.   I also tried binding to a different port, but that gave the same error.  

I'm guessing this is caused by a build environment update, is there anything I can do to work around the issue?

Build Example: https://travis-ci.org/jrjohnson/frontend/builds/41286719

**Comment by steve8708** [comment]
+1 :)

**Comment by Spy-Seth** [comment]
@BanzaiMan, any news on this issue ?

**Comment by BanzaiMan** [comment]
Sorry. We have no updates at this time.

**Comment by shouze** [comment]
@BanzaiMan any news?

**Comment by BanzaiMan** [comment]
No updates. Sorry.

**Comment by guedressel** [comment]
Just 2 cents for you since I stumbled over this issue: Using ```sed``` to edit /etc/hosts will fail since it does a rename in the background. Docker doesn't allow renaming/moving of /etc/hosts.

**Comment by BanzaiMan** [comment]
@guedressel Thank you for the idea! I think that was just the trick we need.

**Comment by fommil** [comment]
:+1: thanks for fixing it

**Comment by damianobarbati** [comment]
Is the host addon working? Sudo:true|false gives the same: no output for host resolution.

```
language: node_js
node_js: v8.1.2
git:
    depth: 1
    submodules: false
addons:
    hosts:
        - dev.quant01.com
        - stg.quant01.com
        - www.quant01.com
    ssh_known_hosts:
        - stg.quant01.com
        - www.quant01.com
install: true
before_script:
    - sudo apt-get install libcap2-bin
    - sudo setcap 'cap_net_bind_service=+ep' `which node`
script:
   - host www.quant01.com # <= here I should get the resolved ip to a local ip but no output so far
```

**Comment by BanzaiMan** [comment]
The "hosts" addon works by manipulating `/etc/hosts`. As `hosts` does not consult this file, `host www.quant01.com` will *_always_* resolve to the DNS name. https://serverfault.com/questions/498500/why-does-the-host-command-not-resolve-entries-in-etc-hosts

**Comment by damianobarbati** [comment]
@BanzaiMan is there anyway to achieve what I was thinking?
Resolve www to local machine and make some tests?

**Comment by BanzaiMan** [comment]
It depends on _how_ you connect to the `www`. If your tool consults `/etc/hosts` (or if you can advise the tool to consult the file), you'll be OK. If this is not an option, you'll have to set up a local DNS server and point `/etc/nsswitch.conf` to it.

**Comment by BanzaiMan** [comment]
Needless to say, the local DNS server resolves `www` to 127.0.0.1.
